src/StoragelessSession/Session/Data.php

```diff
@@ -33,14 +33,50 @@ class Data implements \JsonSerializable
     private $metadata;
 
     /**
+     * @var bool
+     */
+    private $dataWasChanged = false;
```

Comment:
You're exposing it publicly following a "convention".
I'm talking about both - attribute and method - names.

And it's not "wrong" it's just "strange" to ask if the data **inside** the container is changed when we want to know if the container has changes. It's subtle I know but can improve the understandability.